File: app/src/main/java/site/rakshith/myapplication/adapter/OnBoardAdapter.java
package site.rakshith.myapplication.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.PagerAdapter;

import java.util.ArrayList;

import site.rakshith.myapplication.OnBoarding;
import site.rakshith.myapplication.R;
import site.rakshith.myapplication.pojo.OnboardItem;

public class OnBoardAdapter extends PagerAdapter {

    Context context;
    ArrayList<OnboardItem> items;


    public OnBoardAdapter(Context context, ArrayList<OnboardItem> items){
        this.context = context;
        this.items = items;
    }


    @Override
    public int getCount() {
        return items.size();
    }


    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return false;
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {

        View itemView = LayoutInflater.from(context).inflate(R.layout.onboard_viewpager_layout,container,false);

        OnboardItem item = items.get(position);

        itemView.setBackgroundColor(item.getColor());

        ImageView onbordingImage = itemView.findViewById(R.id.onboardImg);
        onbordingImage.setImageResource(item.getImage());

        TextView header = itemView.findViewById(R.id.descriptionHeader);
        header.setText(item.getHeader());

        TextView description = itemView.findViewById(R.id.descriptionBody);
        description.setText(item.getDescription());

        container.addView(itemView);

        return itemView;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        container.removeView((FrameLayout) object);
    }
}

File: app/src/main/java/site/rakshith/myapplication/adapter/OnBoardAdapter.kt
package site.rakshith.myapplication.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import site.rakshith.myapplication.R
import site.rakshith.myapplication.pojo.OnboardItem

class OnBoardAdapter(
    private val context :Context,
    private val items: ArrayList<OnboardItem>
)
    : RecyclerView.Adapter<OnBoardAdapter.OnBoardViewHolder>()
{
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OnBoardViewHolder {
        return OnBoardViewHolder(LayoutInflater.from(context).inflate(R.layout.onboard_viewpager_layout, parent, false)
        )
    }

    override fun getItemCount(): Int {
        return items.size
    }

    override fun onBindViewHolder(holder: OnBoardViewHolder, position: Int) {
        var onboardItem :OnboardItem? = items.get(position)
        if (onboardItem != null) {
            holder.onboardImg?.setImageResource(onboardItem.image)
            holder.descriptionHeader?.setText(onboardItem.header)
            holder.descriptionBody?.setText(onboardItem.description)
        }


    }

    class OnBoardViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var onboardImg : ImageView? = null
        var descriptionHeader : TextView? = null
        var descriptionBody : TextView? = null

        init {
            onboardImg = itemView.findViewById(R.id.onboardImg)
            descriptionHeader = itemView.findViewById(R.id.descriptionHeader)
            descriptionBody = itemView.findViewById(R.id.descriptionBody)

        }
    }
}